### DISCUSSION TITLE
Pocketbase with nextjs 14 and auth

### DISCUSSION THREAD

**Discussion by nicholsss** [pocketbase/pocketbase#5827] [Q&A]
Hello, I've been skimming through discussions and saw some conversations about using auth with Next.js. For example, I found this: https://github.com/pocketbase/js-sdk/issues/69#issuecomment-1840433737. Should I go with a solution like this, or are there other options? I'm interested in trying out PocketBase, but implementing auth with it seems like a heavier task compared to using Next-Auth.

**Comment by ganigeorgiev** [comment]
I cannot advice you about Next.js as I don't have much recent experience with it but I recommend to check https://github.com/pocketbase/pocketbase/discussions/5313.

**Comment by skellgreco** [comment]
Hey! I wrote a nextjs - pocketbase OAuth implementation on this repository to make this task easier! Feel free to check it out:

https://github.com/skellgreco/pocketbase-oauth2-nextjs

**Reply by ganigeorgiev** [reply]
Your `initializePocketBase` is vulnerable if used in a SSR context because you are instantiating the PocketBase instance outside of it, which means that requests from different users will share/overwrite the same `pb.authStore` object. Please read https://github.com/pocketbase/pocketbase/discussions/5313. I'm tired of seeing this bug in the wild and security researchers spamming me about it. If you don't understand what actually the described problems in the linked discussion means (_I can't explain it better_), please don't use PocketBase with JS SSR.
